src/Reader.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Marcosh\LamPHPda;
+
+use Marcosh\LamPHPda\HK\HK1;
+use Marcosh\LamPHPda\Instances\Reader\ReaderApplicative;
+use Marcosh\LamPHPda\Instances\Reader\ReaderApply;
+use Marcosh\LamPHPda\Instances\Reader\ReaderFunctor;
+use Marcosh\LamPHPda\Instances\Reader\ReaderMonad;
+use Marcosh\LamPHPda\Typeclass\Applicative;
+use Marcosh\LamPHPda\Typeclass\Apply;
+use Marcosh\LamPHPda\Typeclass\DefaultInstance\DefaultMonad;
+use Marcosh\LamPHPda\Typeclass\Functor;
+use Marcosh\LamPHPda\Typeclass\Monad;
+use Marcosh\LamPHPda\Brand\ReaderBrand;
+
+/**
+ * @template E
+ * @template A
+ * @implements DefaultMonad<ReaderBrand, A>
+ *
+ * @psalm-immutable
+ */
+final class Reader implements DefaultMonad
+{
+    /**
+     * @var callable(E): A
+     */
+    private $action;
+
+    /**
+     * @param callable(E): A $f
+     *
+     * @psalm-mutation-free
+     */
+    private function __construct(callable $f)
+    {
+        $this->action = $f;
+    }
+
+    /**
+     * @template B
+     * @param callable(E): B $f
+     * @return Reader<E, B>
+     *
+     * @psalm-pure
+     */
+    public static function reader(callable $f): Reader
+    {
+        return new self($f);
+    }
+
+    /**
+     * @template B
+     * @param callable(E): B $b
+     * @return Reader<E, B>
+     *
+     * @psalm-pure
+     */
+    public static function of($b): Reader
+    {
+        return new self(static fn () => $b);
+    }
+
+    /**
+     * @template B
+     * @param HK1<ReaderBrand, B> $b
```

Comment:
also here it'd be useful to use `ReaderBrand<E>`, to that the information about the environment type could be propagated

Reply:
Fixed.